src/runtime-pmi.c

```diff
@@ -77,11 +77,13 @@ decode(const char *inval, void *outval, int outvallen)
     int i;
     char *ret = (char*) outval;
 
-    if (outvallen != strlen(inval) / 2) {
+    if (outvallen < strlen(inval) / 2) {
         return 1;
     }
 
-    for (i = 0 ; i < outvallen ; ++i) {
+    size_t outlen = strlen(inval) / 2;
```

Comment:
Could you move this up and change the conditional to `if (outvallen < outlen)`?  Also, I'm assuming that the NULL termination covered during this horrifying hex->ASCII conversion?